test/services/TariffApiServiceSpec.scala

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import scala.concurrent.ExecutionContext.global
+import scala.concurrent.Future
+
+import base.UnitWithMocksSpec
+import config.featureFlags.TariffApiConfig
+import connectors.TariffApiConnector
+import forms.declaration.CommodityDetails
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito._
+import org.scalatest.{BeforeAndAfterEach, OptionValues}
+import org.scalatest.concurrent.ScalaFutures
+import play.api.libs.json.Json
+import services.cache.{ExportsDeclarationBuilder, ExportsItemBuilder}
+
+class TariffApiServiceSpec
+    extends UnitWithMocksSpec with BeforeAndAfterEach with ExportsDeclarationBuilder with ExportsItemBuilder with OptionValues with ScalaFutures {
+
+  private val tariffApiConfig = mock[TariffApiConfig]
+  private val tariffApiConnector = mock[TariffApiConnector]
+
+  private val tariffApiService = new TariffApiService(tariffApiConfig, tariffApiConnector)(global)
+
+  override def beforeEach(): Unit = {
+    reset(tariffApiConfig, tariffApiConnector)
+
+    when(tariffApiConfig.isCommoditiesEnabled).thenReturn(true)
+    when(tariffApiConnector.getCommodity(any())).thenReturn(Future.successful(None))
+  }
+
+  private val commodityCode = "2208303000"
+
+  private def retrieveCommodityInfoIfAny(code: Option[String] = Some(commodityCode)): Future[Option[CommodityInfo]] = {
+    val commodityDetails = CommodityDetails(code, Some("Description"))
+    val item = anItem(withCommodityDetails(commodityDetails))
+    tariffApiService.retrieveCommodityInfoIfAny(aDeclaration(withItems(item)), item.id)
+  }
+
+  private def retrieveCommodityInfoIfAny(json: String): Option[CommodityInfo] = {
+    val response = Future.successful(Some(Json.parse(json)))
+    when(tariffApiConnector.getCommodity(any())).thenReturn(response)
+    retrieveCommodityInfoIfAny().futureValue
+  }
```

Comment:
Nit: could these two methods be named slightly differently from each other? Overriding the same name makes the code that calls them a bit confusing (which version is being called).